go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -230,39 +229,38 @@ func (ovn *Controller) createLoadBalancerRejectACL(lb string, sourceIP string, s
 		klog.Errorf("Error while querying ACLs by name: %s, %v", stderr, err)
 	} else if len(aclUUID) > 0 {
 		klog.Infof("Existing Service Reject ACL found: %s for %s", aclUUID, aclName)
-		// If we found the ACL exists we need to ensure it applies to all logical switches
-		cmd := []string{}
-		for _, ls := range switches {
-			cmd = append(cmd, "--", "add", "logical_switch", ls, "acls", aclUUID)
-		}
-		if len(cmd) > 0 {
-			_, _, err = util.RunOVNNbctl(cmd...)
-			if err != nil {
-				klog.Warningf("Unable to add reject ACL: %s for switches: %s", aclUUID, switches)
-			}
+		_, stderr, err = util.RunOVNNbctl("--", "add", "port_group", ovn.clusterPortGroupUUID, "acls", aclUUID)
+		if err != nil {
+			klog.Errorf("Failed to add LB %s ACL %s %q to cluster port group: %q (%v)",
+				lb, aclUUID, aclName, stderr, err)
+			return "", err
 		}
+
 		ovn.setServiceACLToLB(lb, vip, aclUUID)
+
+		// If reject ACL exist, ensures that the _uuid is removed from logical_switch acls list.
+		// This step is required to ensure the clean-up when ovn upgrades from logical_switch acls
+		// to port_group based acls.
+		ovn.removeACLFromNodeSwitches(switches, aclUUID)
```

Comment:
this should be handled in the sync method I think.

Reply:
yeah this is a bit tricky.  I guess this is OK since it's only when cache is invalid.